src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -186,6 +194,12 @@ private static void ParseData(string resxFilename, bool pathsRelativeToBasePath,
                     case BinSerializedObjectMimeType:
                     case Beta2CompatSerializedObjectMimeType:
                     case CompatBinSerializedObjectMimeType:
+                        // Warn of BinaryFormatter exposure (on by default in .NET 8+)
+                        if (logWarningForBinaryFormatter)
+                        {
+                            log.LogWarningWithCodeFromResources("GenerateResource.BinaryFormatterUse", name, resxFilename, mimetype);
```

Comment:
Check `log is null`?

Reply:
I'd initially intended it to just never be null, but then I broke my own invariant when tests came along and made a narrower and more confusing invariant. I'll change this.